python/templates/macros/implementations.jinja2

```diff
@@ -19,7 +19,17 @@
 }
 
 Mutable{{ type }} {{ full_type }}::clone() const {
-  return Mutable{{ type }}(podio::utils::MaybeSharedPtr(new {{ type }}Obj(*m_obj), podio::utils::MarkOwned));
+  auto tmp = Mutable{{ type }}(podio::utils::MaybeSharedPtr(new {{ type }}Obj(*m_obj), podio::utils::MarkOwned));
+{% for relation in multi_relations %}
+  // If the current object has been read from a file, then the object may only have a slice of the relation vector
+  // so this slice has to be copied in case we want to modify it
+  if (m_obj->data.{{ relation.name }}_end - m_obj->data.{{ relation.name }}_begin != m_obj->m_{{ relation.name }}->size()) {
+    tmp.m_obj->m_{{ relation.name }} = new std::vector<{{ relation.full_type }}>(m_obj->m_{{ relation.name }}->begin() + m_obj->data.{{ relation.name }}_begin, m_obj->m_{{ relation.name }}->begin() + m_obj->data.{{ relation.name }}_end);
```

Comment:
Are we leaking the `vector` that is created in the `tmp` constructor here by not getting rid of it explicitly? I think we are and the AddressSanitizer should in principle catch this in the unittests, but we would need to go through an I/O backend that has no memory issues. SIO and RNTuple should both work for that. Then we could remove the `ASAN-FAIL` (and hopefully also `UBSAN-FAIL` labels in the tests below.

Reply:
Here I would say that no, or at least that this is the same situation as when a new object is created since then also the vector is created with `new std::vector`. When cloning a collection that is read, the initial big vector is still managed by the read collection and its objects and the new vector by the cloned object.
Sanitizers seem to be happy. Locally all the rntuple tests fail for me with either gcc 13 or clang 17 and ROOT 6.30.06

Reply:
Makes sense. I think the cleanup is correct and since we run it through the sanitizers now this should be OK. I will check again locally to make sure that the tests are actually run.

Reply:
OK. Turns out the cleanup is not OK, and we actually leak the pointer from `tmp`. Looking at the code what happens is the following:
- In the constructor for `tmp` we create a full copy of the big vector (that is managed by the collection)
https://github.com/AIDASoft/podio/blob/43330a8e1ea938f2c98b6473389bf317a6590ec3/python/templates/Obj.cc.jinja2#L32-L37
- In order to only get the slice that we want we create another vector with `new` (this snippet), but do not get rid of the copy we made in the first place

So we need to `delete` the original vector in `tmp` before we put in the new copy of the slice.

Reply:
As far as I can tell from a quick look the `Obj(Obj const&)` copy constructor is only called via `clone`, so we could actually be a bit more clever with the unnecessary copying and only do the necessary work in `clone`.